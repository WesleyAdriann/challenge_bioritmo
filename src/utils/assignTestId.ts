type componentType = 'div' | 'text' | 'input' | 'button' | 'img'

export const assignTestId = (
  componentType: componentType,
  componentId: string
)  => ({
    'data-testid': `${componentType}_${componentId}`,
  })


export const mockTestID = (
    componentType: componentType,
    componentId: string
  ) => assignTestId(componentType, componentId)
