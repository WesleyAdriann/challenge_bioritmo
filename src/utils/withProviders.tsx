
import React from 'react'

import { ThemeProvider } from 'styled-components'

import { theme } from '../style'

export const StorybookProviders: React.FC = ({
  children,
}) => (
  <ThemeProvider theme={theme}>
    {children}
  </ThemeProvider>
)
