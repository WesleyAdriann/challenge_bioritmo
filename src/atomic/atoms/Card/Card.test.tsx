import React from 'react'
import { render, cleanup } from '@testing-library/react'

import { mockTestID, StorybookProviders as wrapper } from '../../../utils'

import { Card, ICard } from './index'

describe('Atoms/Button', () => {
  const props: ICard = {
    testID: 'Card',
    children: 'Text Test'
  }

  const renderComponent = () => render(<Card {...props} />, { wrapper })

  let component = renderComponent()

  beforeEach(() => {
    jest.clearAllMocks()
    cleanup()
    component = renderComponent()
  })

  const el_container = mockTestID('div', props.testID!)

  it(`should render component #${el_container}`, async () => {
    const sut = await component.findByTestId(el_container)
    expect(sut).toBeTruthy()
  })

  it('should render Card with gray background-color', async () => {
    const color = 'gray'
    component.rerender(<Card {...props} color={color} />)
    const sut = await component.findByTestId(el_container)
    expect(sut).toHaveStyle(`background-color: ${color}`)
  })

  it('should render snapshot', () => {
    expect(component.container.firstChild).toMatchSnapshot()
  })
})
