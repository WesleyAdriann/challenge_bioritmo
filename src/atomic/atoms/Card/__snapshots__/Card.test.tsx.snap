// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/Button should render snapshot 1`] = `
.c0 {
  background-color: #F3F3F3;
  border-radius: 0px;
  box-shadow: 0px;
  border: 0px solid #F3F3F3;
  padding: 0px;
}

<div
  class="c0"
  data-testid="div_Card"
>
  Text Test
</div>
`;
